src/query/scan_manager.c

```diff
@@ -7852,28 +7877,53 @@ scan_build_hash_list_scan (THREAD_ENTRY * thread_p, SCAN_ID * scan_id)
 	  new_key = key;
 	}
 
+      /* make hash key */
+      hash_key = qdata_hash_scan_key (new_key, UINT_MAX);
+
       /* create new value */
       if (llsidp->hlsid.hash_list_scan_yn == HASH_METH_IN_MEM)
 	{
 	  new_value = qdata_alloc_hscan_value (thread_p, tplrec.tpl);
+	  if (new_value == NULL)
+	    {
+	      return S_ERROR;
+	    }
 	}
       else if (llsidp->hlsid.hash_list_scan_yn == HASH_METH_HYBRID)
 	{
 	  new_value = qdata_alloc_hscan_value_OID (thread_p, &llsidp->lsid);
+	  if (new_value == NULL)
+	    {
+	      return S_ERROR;
+	    }
+	}
+      else if (llsidp->hlsid.hash_list_scan_yn == HASH_METH_HASH_FILE)
+	{
+	  /* curr_offset is int and slotid is short. */
+	  /* In fact, the offset is a position within a page(16K), so it can be stored as a short type. */
+	  oid.volid = llsidp->lsid.curr_vpid.volid;
+	  oid.pageid = llsidp->lsid.curr_vpid.pageid;
+	  oid.slotid = llsidp->lsid.curr_offset;
```

Comment:
How about defining a new structure for hash ID instead of resuing OID with dummy `slotid`.
It could make confusion with the real OID.

```cpp
// for example
struct HFSID
{
  int volid;
  int pageid;
  int offset;
}
```

Reply:
I fixed it.